### DISCUSSION TITLE
Combining different language versions to create a multi-language deck

### DISCUSSION THREAD

**Discussion by ErinSolstice** [axelboc/anki-ultimate-geography#621] [Q&A]
Is there an easy way to combine fields from multiple translations to create a deck which shows the Country and Capital fields in multiple languages? For example:

- Country: Germany, Alemania, Deutschland
- Capital: Berlin, Berlín, Berlin

I´m currently learning German and already know both English and Spanish. I've been studying the Spanish version of the deck but I'd also like to do so for both German and English as well. While I could download each of the decks separately, that would significantly increase the number of cards I'd have to study. I also don't really need to relearn the location and flag of say Lettland. I just need to associate it with its English name of Latvia.

I don't mind having to do a bit of tinkering to make this work. But given my unfamiliarity with how the deck building process of AUG works, I figured it'd be best to first ask if y'all had any thoughts on the best way of doing it. Or if it would simply be more trouble than its worth?

**Comment by aplaice** [comment]
I don't know how much tinkering you have in mind!

You'd need to:

1. adjust the "recipe" that BrainBrew uses to generate the CrowdAnki JSON files (changing/copying [this](https://github.com/anki-geo/ultimate-geography//blob/master/recipes/source_to_anki.yaml).)
2. create a custom note type ("note model") (changing the [YAML description of the note model](https://github.com/anki-geo/ultimate-geography//blob/master/src/note_models/Ultimate_Geography.yaml) and adding a couple of [new HTML templates](https://github.com/anki-geo/ultimate-geography/tree/master/src/note_models/templates) (one for each new card type (say "Multilingual Country" or "Multilingual Capital")).

(Approximately: the first mainly manages how the columns in the source CSVs (`src/data/*.csv`) are mapped onto the fields in the target notes, while the second describes how the fields in the Anki notes map onto the Anki cards.)

Our "build/import" process is as follows:


```mermaid
graph LR;
    A[src/*] -- BrainBrew --> B[CrowdAnki JSON];
    B -- CrowdAnki --> C[Anki deck];
```

where the first part is usually run by us (to generate the "released" CrowdAnki decks/JSON files) and the second part usually run by the user. In this case you'll need to run BrainBrew yourself (see https://github.com/anki-geo/ultimate-geography/blob/master/CONTRIBUTING.md#getting-started).

Unfortunately, BrainBrew, while extremely flexible and generally amazing, has slightly complicated recipe files, so please do ask if something is not entirely clear!

**Comment by axelboc** [comment]
> While I could download each of the decks separately, that would significantly increase the number of cards I'd have to study. I also don't really need to relearn the location and flag of say Lettland. I just need to associate it with its English name of Latvia.

Just adding my two cents here: the solution of importing the decks separately (and therefore keeping their respective cards separate) has the advantage of also keeping your progress separate for each language - i.e. if you add a new language later, then the new cards will be identified as "new", independently from all the other cards you've already learnt. If you merge all the languages together into the same cards, then when you add a new language, the content of the cards will be updated but their progresses will not be reset. Maybe that's what you want but I thought I'd point it out just in case. :grin: 

I should also mention that the specific problem of the duplicated map/flag cards can be solved in multiple ways:

- by combining all your UG decks into a [filtered deck](https://github.com/anki-geo/ultimate-geography#custom-study), while also excluding the map/flag cards of all decks except one;
- by removing the map/flag note types of all decks except one (unless the UG note type is shared between all the decks, I can't remember);
- by suspending all the map/flag cards of all decks except one;
- ... (maybe more)

**Reply by aplaice** [reply]
I agree that your solutions are probably the simplest — I just wanted to show how to have multilingual cards if you really want them. :)

> If you merge all the languages together into the same cards, then when you add a new language, the content of the cards will be updated but their progresses will not be reset. Maybe that's what you want but I thought I'd point it out just in case. 😁

I think if you do something like:

1. Have the 6 standard cards for language A.
2. Add, say, 2 new cards for language B (e.g. Country in A -> Country in B; and Capital in A -> Capital in B) (you can have more combinations.
3. Add 2 new cards for language C (e.g. Country A->C and Capital A->C)
4. same for D etc.

(i.e. if you only add cards to the note type or at least never add anything "load-bearing" to existing cards) then if you're careful, you won't have any issues with "incorrect" progress.

> unless the UG note type is shared between all the decks, I can't remember

Unfortunately, it is. The end-user can clone and change the note type for the relevant notes, though.

**Reply by axelboc** [reply]
> I just wanted to show how to have multilingual cards if you really want them.

For sure, @ErinSolstice asked for the hands-on approach :+1: 

> Add, say, 2 new cards for language B (e.g. Country in A -> Country in B; and Capital in A -> Capital in B) (you can have more combinations.

If I'm not mistaken, @ErinSolstice has in mind the following template(s): _Country in A/B <-> Capital in A/B_ — i.e. all the studied languages listed together in the same _Country <-> Capital_ cards.

So if you add language C later on, the template(s) becomes _Country in A/B/C <-> Capital in A/B/C_ and only the contents of the cards change when the deck is re-imported into Anki; no cards are removed/created. As a result, all the cards that have already been reviewed with languages A and B are not reset as "new", which makes learning language C challenging.
